eng/ApiCompat/ApiCompat.csproj

```diff
@@ -1,41 +1,31 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
-  <!--This project is used to enforce API compatibility between the GA'ed libraries and the most recent version available on Nuget. The most recent
+  <!--
+  This project is used to enforce API compatibility between the GA'ed libraries and the most recent version available on Nuget. The most recent
   version is restored from Nuget via the package references below. The ApiCompatVerification target, specified below, is referenced in the
-  eng/Directory.Build.Data.targets file which causes this target to be executed for each csproj that has the EnableApiCompat parameter set to true.
-  The EnableApiCompat parameter defaults to the value of the IsShippingClientLibrary parameter, which is defined in the eng/Directory.Build.Data.props file.-->
+  eng/Directory.Build.Data.targets file which causes this target to be executed for each csproj that has an ApiCompatVersion set.
+  -->
   <PropertyGroup>
     <TargetFramework>netstandard2.0</TargetFramework>
     <RunApiCompatForSrc>true</RunApiCompatForSrc>
-    <IntermediateOutputPath>$(IntermediateOutputPath)/$(TargetPackageName)</IntermediateOutputPath>    
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.DotNet.ApiCompat" />
+    <PackageReference Include="Microsoft.DotNet.ApiCompat" Version="5.0.0-beta.19552.1" />
+    <PackageReference Include="$(TargetPackageName)" Version="$(TargetPackageVersion)" />
```

Comment:
How does this work in VS when you have multiple projects open?

Reply:
This project should never really be loaded in VS but you bring up a good question that I should verify that these changes work in VS which I haven't done yet.

Reply:
We had problems with building other projects when this one is not loaded in VS.

Currently it's added to all solutions.

Reply:
Also, does this work well without requiring a manual restore?

Reply:
The manual restore is not at all needed with this solution, as we restore as part of the call to the run the apicompat target. We no longer have a projectreference to this project.

Reply:
OK I just did some basic testing in VS (used keyvault solution for testing) and while the APICompat project is in the solution it will build but will not end up doing anything interesting when building without the extra properties, it doesn't fail but it also doesn't do anything interesting. The ApiCompat checks still work as expected for the other projects when building in VS.

So in the end things appear to work as I expect and we should just remove the ApiCompat project from the solutions. I'll make a pass over those and clean those up as part of this PR.

Reply:
Last question, how's the build time considering we do additional restore per every project?

Reply:
I haven't looked at before and after build times but I don't think it will be much of an impact because the Restore step should be a no-op after the first time. I will take a look at a core build to see if there is any notice impact.

Reply:
I looked at a few random builds before and after my change of the core pipeline and the build and package are both in the 7min range so nothing really jumps out. However with that said please do let me know if anyone sees any noticeable performance issues.